src/index.ts

```diff
@@ -154,7 +165,10 @@ class Upload {
       payload => {
         if (payload.uploadId === this.uploadId) {
           this.url = payload.uploadUrl;
-          this.onSuccess();
+          if (!this.aborting) {
```

Comment:
This logic will introduce more problems. Since an upload can be aborted and resumed, we might end up in a situation where the upload has been aborted and resumed by the user, but the `onSuccess` never called.

I would remove the `aborting` flag.